template.php

```diff
@@ -169,6 +199,15 @@ function campaignion_foundation_block_view_alter(&$data, $block) {
       $link['attributes']['class'][] = $link['attributes']['data-share'] . '-icon';
     }
   }
+  // Remove disabled links from campaignion language switcher blocks.
+  if ($block->module == 'campaignion_language_switcher') {
```

Comment:
I think this should be possible in `campaignion_foundation_preprocess_campaignion_language_switcher()` as well.

Reply:
Unfortunately it is not, the links all still have a `href` attribute it the preprocess function, that gets only later removed by `hook_block_view()`.

Reply:
Are you sure?

AFAIK:  `campaignion_language_switcher_block_view()` calls `language_negotiation_get_switch_links()` which in turn invokes `hook_language_switch_links_alter()` thereby also `campaignion_language_switcher_language_switch_links_alter()` which removes the `href`. The result of `campaignion_language_switcher_block_view()` is a renderable array so the theme function is not removed until after the function has returned. So the `href` should already be removed when the theme-layer becomes active.

Reply:
Yes. After `campaignion_language_switcher_block_view()` we have the final renderable array as you described, that’s why `campaignion_foundation_block_view_alter()` works. The theme function does its thing before any of that, not after as I would have expected too.

Reply:
That would be crazy. Are you sure? It would mean that `theme()` is run before `render()` is run on the same data structure.

Reply:
It looks that way.